### ISSUE TITLE
Make utils-tests-helpers `overrides` strongly typed

### ISSUE THREAD

**Issue by antross** [webhintio/hint#3956]
If we want to get really fancy I may be able to use generics in the type definition of `overrides` to derive this type automatically based on the referenced module name. Which would be cool since it'd flag typos or signature mis-matches versus the actual module. I feel it's out of scope here since we're trying to unblock a dependency update, but could be neat to explore in the future.

_Originally posted by @antross in https://github.com/webhintio/hint/pull/3899_

**Comment by molant** [comment]
Adding to agenda so it can be tagged accordingly.
